test/e2e/operator/garden/create_delete.go

```diff
@@ -5,185 +5,117 @@
+				controllerRegistrationList := &gardencorev1beta1.ControllerRegistrationList{}
+				Eventually(ctx, s.VirtualClusterKomega.List(controllerRegistrationList)).Should(Succeed())
+				Expect(controllerRegistrationList.Items).To(ContainElement(MatchFields(IgnoreExtras, Fields{"ObjectMeta": MatchFields(IgnoreExtras, Fields{"Name": Equal("provider-local")})})))
+
+				controllerDeploymentList := &gardencorev1.ControllerDeploymentList{}
+				Eventually(ctx, s.VirtualClusterKomega.List(controllerDeploymentList)).Should(Succeed())
+				Expect(controllerDeploymentList.Items).To(ContainElement(MatchFields(IgnoreExtras, Fields{"ObjectMeta": MatchFields(IgnoreExtras, Fields{"Name": Equal("provider-local")})})))
+			}, SpecTimeout(time.Minute))
+
+			It("Verify 'gardener-system-public' namespace and 'gardener-info' configmap exist", func(ctx SpecContext) {
+				namespace := &corev1.Namespace{
+					ObjectMeta: metav1.ObjectMeta{
+						Name: gardencorev1beta1.GardenerSystemPublicNamespace,
+					},
+				}
+				Eventually(ctx, s.VirtualClusterKomega.Get(namespace)).Should(Succeed())
+
+				configMap := &corev1.ConfigMap{
+					ObjectMeta: metav1.ObjectMeta{
+						Name:      "gardener-info",
+						Namespace: gardencorev1beta1.GardenerSystemPublicNamespace,
+					},
+				}
+				Eventually(ctx, s.VirtualClusterKomega.Get(configMap)).Should(Succeed())
+				Expect(configMap.Data).To(HaveKey("gardenerAPIServer"))
+			}, SpecTimeout(time.Minute))
+
+			ItShouldDeleteGarden(s)
+			ItShouldWaitForGardenToBeDeleted(s)
+			ItShouldCleanUp(s)
+
+			It("Verify no leftover secrets", func(ctx SpecContext) {
+				secretList := &corev1.SecretList{}
+				Eventually(ctx, s.GardenKomega.List(secretList, client.InNamespace(v1beta1constants.GardenNamespace), client.MatchingLabels{
+					secretsmanager.LabelKeyManagedBy:       secretsmanager.LabelValueSecretsManager,
+					secretsmanager.LabelKeyManagerIdentity: operatorv1alpha1.SecretManagerIdentityOperator,
+				})).Should(Succeed())
+
+				Expect(secretList.Items).To(BeEmpty())
+			}, SpecTimeout(time.Minute))
+
+			It("Verify CRD still present", func(ctx SpecContext) {
+				crdList := &apiextensionsv1.CustomResourceDefinitionList{}
+				Eventually(ctx, s.GardenKomega.List(crdList)).Should(Succeed())
+				Expect(crdList.Items).To(ContainElement(MatchFields(IgnoreExtras, Fields{"ObjectMeta": MatchFields(IgnoreExtras, Fields{"Name": Equal("gardens.operator.gardener.cloud")})})))
+			}, SpecTimeout(time.Minute))
+
+			It("Verify gardener-resource-manager was deleted", func(ctx SpecContext) {
+				deployment := &appsv1.Deployment{
+					ObjectMeta: metav1.ObjectMeta{
+						Name:      v1beta1constants.DeploymentNameGardenerResourceManager,
+						Namespace: v1beta1constants.GardenNamespace,
+					},
+				}
+
+				Eventually(ctx, s.GardenKomega.Get(deployment)).Should(BeNotFoundError())
+			}, SpecTimeout(time.Minute))
+
+			ItShouldWaitForExtensionToReportDeletion(s, "provider-local")
+		}
+
+		Context("Garden", Ordered, func() {
```

Comment:
Do we need this context and the `test` function at all? There is only one case anyways?

Reply:
We don't necessarily need the context.
I only added it because all the other tests also utilized contexts.

If you want to, I can remove it 😊

Reply:
removed it 👍